labrador/src/zq.rs

```diff
@@ -0,0 +1,180 @@
+use std::fmt;
+use std::ops::{Add, AddAssign, Mul, MulAssign, Sub, SubAssign};
+/// Represents an element in the ring Z/qZ where q = 2^32.
+/// Uses native u32 operations with automatic modulo reduction through wrapping arithmetic.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub struct Zq {
+    /// Stored value is always in [0, q-1] due to u32's wrapping behavior
+    value: u32,
+}
+
+impl Zq {
+    /// Modulus q = 2^32 (stored as 0 in u32 due to wrapping behavior)
+    pub const Q: u32 = u32::MAX.wrapping_add(1);
+
+    /// Creates a new Zq element from a raw u32 value.
+    /// No explicit modulo needed as u32 automatically wraps
+    pub fn new(value: u32) -> Self {
```

Comment:
In Rust (as well as in C++ constexpt, Zig, Nim etc) we have such a nice feature as compile time evaluation (`const`). Actually, the compiler may evaluate the `const fn` at compile time (if called in a const context) rather than at runtime, reducing the number of instructions.

For example https://godbolt.org/z/rx9s8fEs4 (try to uncomment each line in `main` in turn to see the resulting assembly)

You can also replace `Zq::new(1)` with `Zq::ONE`  in the tests to play around :)

Reply:
Thanks! Good to know!